tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -240,8 +244,7 @@ private[emitter] class JSDesugaring(semantics: Semantics,
   private[emitter] def transformParamDef(paramDef: ParamDef): js.ParamDef =
     js.ParamDef(paramDef.name, paramDef.rest)(paramDef.pos)
 
-  private class JSDesugar(enclosingClassName: String)(
-      implicit globalKnowledge: GlobalKnowledge) {
+  private class JSDesugar(implicit globalKnowledge: GlobalKnowledge) {
```

Comment:
Keep an explicit empty pair of `()` in the constructor. Scala adds them implicitly for a constructor that has only an implicit parameter list, but IMO they should be explicit.